clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -322,23 +333,26 @@ private boolean maybeSendTransactionalRequest(long now) {
                 }
                 if (targetNode != null) {
                     if (nextRequestHandler.isRetry()) {
+                        log.trace("waiting " + retryBackoffMs + "ms before resending a transactional request: " + nextRequestHandler.requestBuilder());
```

Comment:
nit: seems we usually capitalize the first word in log messages. Would be nice to be consistent. Also, we should use the `{}` substitutions.